File: backend-dotnetcore/Models/Movie.cs
using System.ComponentModel.DataAnnotations;

namespace RUG.WebEng.Movies.Models;

public class Movie
{
    [Key] // Indicates this is the primary key of the table
    public int Id { get; set; }

    [Required]
    public string Title { get; set; }
    public string? Rating { get; set; } // Could possibly be an enum
    public int? Year { get; set; }
    public string? Description { get; set; }
    public Review Review { get; set; } = new Review();
    public ICollection<Actor> Actors { get; set; } = new List<Actor>();
    public ICollection<Language> Languages { get; set; } = new List<Language>();

    [Required]
    public string IMDbUrl { get; set; }
}

File: backend-express/models/movie.ts
import { Actor } from "./actor";
import { Language } from "./language";
import { Review } from "./review";
import { Entity, Column, PrimaryGeneratedColumn, ManyToMany, JoinTable } from "typeorm";

@Entity({ name: "Movies" })
export class Movie {
    @PrimaryGeneratedColumn()
    id: number;

    @Column()
    title: string;

    @Column()
    rating?: string;

    @Column()
    year?: number;

    @Column("text")
    description?: string;

    @Column(() => Review, {
        prefix: "Review_"
    })
    review: Review;

    @ManyToMany(_ => Actor, a => a.movies, { eager: true })
    @JoinTable({
        name: "ActorMovie",
        joinColumn: {
            name: "MoviesId"
        },
        inverseJoinColumn: {
            name: "ActorsId"
        }
    })
    actors: Actor[];

    @ManyToMany(_ => Language, l => l.movies, { eager: true })
    @JoinTable({
        name: "LanguageMovie",
        joinColumn: {
            name: "MoviesId"
        },
        inverseJoinColumn: {
            name: "LanguagesId"
        }
    })
    languages: Language[];

    @Column({ name: "IMDbUrl" })
    imdb_url: string;
}